src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -33,8 +33,13 @@ internal ConfidentialClientApplicationBuilder(ApplicationConfiguration configura
             ConfidentialClientApplicationOptions options)
         {
             var config = new ApplicationConfiguration();
-            return new ConfidentialClientApplicationBuilder(config)
-                   .WithOptions(options).WithClientSecret(options.ClientSecret);
+            var builder = new ConfidentialClientApplicationBuilder(config).WithOptions(options);
```

Comment:
Looks good, but please add a unit test around this.

Reply:
Yup, adding one already.